### ISSUE TITLE
Drift rate parameter 'v' not estimated properly

### ISSUE THREAD

**Issue by Lipika-T** [lnccbrown/HSSM#528]
**Describe the bug**
I am trying to fit a drift diffusion model to a letter search task with 2 target letters (2 choice). There's a conditioning variable - "Emotion" in the dataset with 3 levels and 48 trials per level, and the accuracy in the task is around 95%. I first used correct trials with the 2 target letters as the thresholds and the drift rates are very small (close to 0) and also often negative with large standard deviation. I then tried using all the trials with "correct" and "incorrect" as the 2 thresholds. The z value in this case is not biased towards the correct threshold. The non decision time values also seem large given that the RT values range from around 0.54 to 1.4 seconds in this participant. 

**HSSM version**
'0.2.3'

**To Reproduce**
simple_ddm_model_reg_v = hssm.HSSM(
    data=df,
    model="ddm",
    hierarchical = False,
    include=[
        {
            "name": "v",
            "formula": "v ~ 0 + (1|Emotion)",
        },
        {
            "name": "a",
        },
        {
            "name": "z",
        },
        {
            "name": "t",
        },
    ],
)
**Screenshots**
Example dataset and model parameters from one participant.
![Screenshot from 2024-08-01 11-33-33](https://github.com/user-attachments/assets/26ccb0ca-28dd-4c76-aa05-d76b0aafc70f)
![Screenshot from 2024-08-01 11-39-30](https://github.com/user-attachments/assets/1b6769b5-43cc-4d2c-a123-bde77c553857)
![Screenshot from 2024-08-01 11-49-22](https://github.com/user-attachments/assets/4918d80b-298f-43de-b667-8886633e9976)
![Screenshot from 2024-08-01 11-56-46](https://github.com/user-attachments/assets/e3cdf08c-3aa7-4c24-ac01-3af44a61eb93)

**Comment by SaschaFroelich** [comment]
Hi,

> I then tried using all the trials with "correct" and "incorrect" as the 2 thresholds.

As far as I understand your post, this means you had 95% / 5% response proportions, right? This can be problematic and result in spurious estimates, especially of drift rate and non-decision time. Response proportions should ideally be balanced out, or contain at least 10-15% of the less common response, as far as I know.

Best,
Sascha

**Comment by Lipika-T** [comment]
Yes, that's right. But that should not be a problem when we keep the 2 target letters as the bounds right? Each of the letter occurs 50% of the time. 'model_summary2' has the estimates from the model using these bounds but the estimates don't seem to make sense.

**Comment by frankmj** [comment]
When you make the two bounds the target letters, then you need something
like the equivalent of stimcoding in HDDM. Here the overall drift rate
across all trials of each emotion level should be 0 assuming that you have
equal trials in which the bound should hit each letter. So you would need
to have a stimulus column that tells the model which response should have
been made for that stimulus and have v modeled as a function of that. eg if
you coded stim as simply 1 and -1 then v ~ stim should give positive
coefficients for stim. This can then interact with Emotion condition if you
want.

Also you should not really model Emotion as coming from a hierarchical
distribution but rather just another term (e.g. v~ stim +stim*Emotion (you
can have this modeled hierarchically across subjects, but that is a
separate issue - that would be v ~1 + stim+ stim*Emotion
+(1+stim+stim*Emotion|partipant_id).

In principle one could assume that different emotion conditions within a
subject are themselves drawn from an overarching distribution but if you
only have 3 conditions that is not sufficient to estimate the overarching
sigma (Gelman recommends at least 5 would be needed).

This likely isn't really an "issue" as much as a "discussion"  so we should
probably switch it over on github

M
